Xamarin.Essentials/Contacts/Contacts.android.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Provider;
+using Net = Android.Net;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contacts
+    {
+        static async Task<Contact> PlatformPickContactAsync()
+        {
+            using var intent = new Intent(Intent.ActionPick);
+            intent.SetType(ContactsContract.CommonDataKinds.Phone.ContentType);
+            var result = await IntermediateActivity.StartAsync(intent, 101).ConfigureAwait(false);
+
+            if (result?.Data != null)
+                return PlatformGetContacts(result.Data);
+
+            return null;
+        }
+
+        internal static Contact PlatformGetContacts(Net.Uri contactUri)
+        {
+            if (contactUri == null)
+                return default;
+
+            using var context = Platform.GetCurrentActivity(true).ContentResolver;
+
+            using var cur = context.Query(contactUri, null, null, null, null);
+            var emails = new List<ContactEmail>();
+            var phones = new List<ContactPhone>();
+            var bDate = string.Empty;
+
+            if (cur.MoveToFirst())
+            {
+                var typeOfContact = cur.GetString(cur.GetColumnIndex(ContactsContract.CommonDataKinds.Phone.InterfaceConsts.Type));
+
+                var name = cur.GetString(cur.GetColumnIndex(ContactsContract.Contacts.InterfaceConsts.DisplayName));
+                string id;
+
+                id = cur.GetString(cur.GetColumnIndex(ContactsContract.CommonDataKinds.Email.InterfaceConsts.ContactId));
```

Comment:
Maybe condense this into a single line.

```suggestion
                var id = cur.GetString(cur.GetColumnIndex(ContactsContract.CommonDataKinds.Email.InterfaceConsts.ContactId));
```